addons/sale_loyalty/models/sale_order.py

```diff
@@ -673,10 +673,12 @@ def _get_claimable_rewards(self, forced_coupons=None):
                     continue
                 # Discounts are not allowed if the total is zero unless there is a payment reward, in which case we allow discounts.
                 # If the total is 0 again without the payment reward it will be removed.
-                if reward.reward_type == 'discount' and total_is_zero and (not has_payment_reward or reward.program_id.is_payment_program):
+                is_discount = reward.reward_type == 'discount'
+                is_payment_program = reward.program_id.is_payment_program
+                if is_discount and total_is_zero and (not has_payment_reward or is_payment_program):
                     continue
-                # Skip discount that has already been applied
-                if reward.reward_type == 'discount' and coupon in self.order_line.coupon_id:
+                # Skip discount that has already been applied if not part of a payment program
+                if is_discount and not is_payment_program and reward in self.order_line.reward_id:
```

Comment:
a reward can be a discount and a payment program? :thinking:

Reply:
discount per point, yes =]